### ISSUE TITLE
Gutenberg: error when in the editor

### ISSUE THREAD

**Issue by jeherve** [Automattic/jetpack#10495]
The following error has started to appear at times, when viewing the editor under Posts > Add New:

```
| <b>Warning</b>:  Cannot modify header information - headers already sent by (output started at /srv/users/user9c07aa14/apps/user9c07aa14/public/wp-admin/includes/template.php:2053) in <b>/srv/users/user9c07aa14/apps/user9c07aa14/public/wp-content/plugins/jetpack-dev/class.jetpack.php</b> on line <b>5849</b><br />
```

It would appear that we first noticed the error here:
https://github.com/Automattic/jetpack/pull/10327#pullrequestreview-166924254

**Comment by kraftbj** [comment]
It looks like you stumbled upon something that has existed since https://github.com/Automattic/jetpack/pull/4012 (cc: @dereksmart ) in some ways.

What's happening is `state`, if the static `$path` isn't set, will fire `setcookies` which must be run before any output has been sent to the browser by PHP.

The react page's `get_initial_state` function (which is new, but the functionality has existed since 4012) fires `state`, so it can sometimes set cookies.

For both the React page and Gutenberg, we are calling this function too late.

For the React page, we're calling it on `page_admin_scripts ([#](https://github.com/Automattic/jetpack/blob/6.7-beta/_inc/lib/admin-pages/class.jetpack-react-page.php#L170)) which is firing in (after a jump) via `admin_print_scripts-$hook` ([#](https://github.com/Automattic/jetpack/blob/6.7-beta/_inc/lib/admin-pages/class.jetpack-admin-page.php#L59)).

That hook fires after output has started in Core ([)](https://github.com/WordPress/WordPress/blob/4.9.8/wp-admin/admin-header.php#L118).

For the Gutenberg use case, similar. We are requesting the inital_state function ([#](https://github.com/Automattic/jetpack/blob/6.7-beta/class.jetpack-gutenberg.php#L212)) from within the `enqueue_block_editor_assets` hook ([#](https://github.com/Automattic/jetpack/blob/6.7-beta/class.jetpack.php#L541)), which fires in Core when loading the block editor, after the admin_header has been dropped.

tl;dr: We need to not call `Jetpack::state` within that initial state function if we're going to use it where we're using it.
